core/src/backend/video.rs

```diff
@@ -0,0 +1,140 @@
+//! Video decoder backends
+
+use crate::backend::render::{BitmapInfo, RenderBackend};
+use generational_arena::{Arena, Index};
+use swf::{VideoCodec, VideoDeblocking};
+
+pub type VideoStreamHandle = Index;
+
+type Error = Box<dyn std::error::Error>;
+
+/// An encoded video frame of some video codec.
+pub struct EncodedFrame<'a> {
+    /// The codec used to encode the frame.
+    codec: VideoCodec,
+
+    /// The raw bitstream data to funnel into the codec.
+    data: &'a [u8],
+
+    /// A caller-specified frame ID. Frame IDs must be consistent between
```

Comment:
Looks like this sentence got cut in half during the separation.
Probably not a big deal, just nitpicking - after all, I (personally) don't find it super important that every single commit in a PR like this is valid and makes sense in itself.